InnoClinic.OfficesAPI/DAL/Repositories/OfficeRepoository.cs

```diff
@@ -0,0 +1,37 @@
+﻿using DAL.Context;
+using DAL.Entities;
+using DAL.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using MongoDB.Bson;
+
+namespace DAL.Repositories;
+
+public class OfficeRepoository(ApplicationDbContext dbContext) : IOfficeRepository
+{
+    public async Task<Office> CreateOfficeAsync(Office office, CancellationToken cancellationToken = default)
+    {
+        await dbContext.Offices.AddAsync(office, cancellationToken);
+        await dbContext.SaveChangesAsync(cancellationToken);
+        return office;
+    }
+
+    public async Task<List<Office>> GetOfficesAsync(CancellationToken cancellationToken = default) =>
+        await dbContext.Offices.ToListAsync(cancellationToken);
+
+    public async Task<Office?> GetOfficeByIdAsync(ObjectId id, CancellationToken cancellationToken = default) =>
+        await dbContext.Offices.FindAsync(id, cancellationToken);
```

Comment:
Do you know about `AsNoTracking`?
What it does and what differences in using it with `SaveChanges`?